### ISSUE TITLE
Odd roundabout exit behaviour 

### ISSUE THREAD

**Issue by pordini** [osmandapp/OsmAnd#18279]
### Describe the routing engine used (required)

- [x] OsmAnd's in-app offline routing
- [ ] Any online routing provider (YOURS, openrouteservice, OSRM, etc.)

### Describe your routing Profile (required)

Bicycle

### Describe your start and end points (required)

https://osmand.net/map/?start=52.082127,4.293916&finish=52.082757,4.295375

### Describe the actual route (required)

Take the 6th exit

### Describe the expected route (required)

Take the 2nd exit

### Describe what Maps you used (online or offline) (required)

- [x] Offline maps offered within the OsmAnd app for download. 
- [ ] Online (tile / raster) maps 

### Your Environment (required)

```
OsmAnd Version: OsmAnd~ 4.5.8
Android/iOS version: Android 13
Device model: Pixel 5
```

### Anything else? (optional)

![IMG_20231003_190926_948](https://github.com/osmandapp/OsmAnd/assets/12774050/738f1e08-a499-438d-b82b-78a767441d2f)

**Comment by DmitryAlexei** [comment]
Similar issues: https://github.com/osmandapp/OsmAnd/issues/18093 and https://github.com/osmandapp/OsmAnd/issues/17715

**Comment by pordini** [comment]
Only the roundabout part. 

The 6th exit part is more interesting in the real world, since for a bicycle the footpaths on the inside are not accessible. So it's not the 6th exit but the 2nd one.

> Similar issues: #18093 and #17715

**Comment by DmitryAlexei** [comment]
@pordini yes, it seems, that highway=footway https://www.openstreetmap.org/way/987424999 on the left are also considered as exits
- bicycle routing

https://test.osmand.net/map/?start=52.082127,4.293916&finish=52.082757,4.295375&type=osmand&profile=bicycle#19/52.08243/4.29454

**Comment by vshcherb** [comment]
Probably for bicycle we should ignore footways cause on ordinary roundabouts they also produce confusing exit numbers or we don't display roundabouts at all .

**Comment by vmicho** [comment]
For example, you can count only road exits as exits, then instruct to take s specific minor exit after/before (whichever is closer) that exit.
E.g. "Take the (left)footpath exit (just)before/(just)after/at the 3rd (road) exit"

**Comment by vmicho** [comment]
PS: the road exits without bicycle access (e.g. motorway links, or ways with explicit bicycle=no) are not counted in the exit number. Which is horribly confusing!
